server/lib/object-storage.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  CompletedPart,
+  CompleteMultipartUploadCommand,
+  CreateMultipartUploadCommand,
+  DeleteObjectCommand,
+  DeleteObjectsCommand,
+  GetObjectCommand,
+  ListObjectsV2Command,
+  PutObjectCommand,
+  S3Client,
+  UploadPartCommand
+} from "@aws-sdk/client-s3"
+import { CONFIG } from "@server/initializers/config"
+import { logger } from '@server/helpers/logger'
+import { createReadStream, createWriteStream, ensureDir, open, close, ReadStream, stat, Stats } from "fs-extra"
+import { Readable } from "stream"
+import { pipeline } from "stream/promises"
```

Comment:
`stream/promises` is not available in node 14. Use `pipelinePromise` from `server/helpers/core-utils` instead